deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/FileUploadCompletionNotification.java

```diff
@@ -0,0 +1,195 @@
+     * @param statusDescription is the description of the status code.
+     * @throws IllegalArgumentException if one of the parameters is null, empty, or not valid.
+     */
+    private void updateStatus(Boolean isSuccess, Integer statusCode, String statusDescription) throws IllegalArgumentException
+    {
+        this.isSuccess = isSuccess;
+        this.statusCode = statusCode;
+        this.statusDescription = statusDescription;
+    }
+
+    /**
+     * Convert this class in a valid json.
+     *
+     * @return a valid json that represents the content of this class.
+     */
+    public String toJson()
+    {
+        Gson gson = new GsonBuilder().disableHtmlEscaping().serializeNulls().create();
+
+        return gson.toJson(this);
+    }
+
+    /**
+     * Set the correlationId that correlates this FileUploadCompletionNotification to the earlier request to get the SAS URI
+     * for this upload from IoT Hub. Must equal {@link FileUploadSasUriResponse#getCorrelationId()}.
+     * @param correlationId the unique identifier that correlates this file upload status to a SAS URI that IoT Hub retreived from Azure Storage earlier.
+     */
+    public void setCorrelationId(String correlationId)
+    {
+        this.correlationId = correlationId;
+    }
+
+    /**
+     * @return Get the correlationId that correlates this FileUploadCompletionNotification to the earlier request to get the SAS URI
+     */
+    public String getCorrelationId()
+    {
+        return this.correlationId;
+    }
+
+    /**
+     * @return Get if the file upload was successful
+     */
+    public Boolean getSuccess()
+    {
+        return this.isSuccess;
+    }
+
+    /**
+     * Set if the file upload was a success
+     * @param success true if the file upload was a success. False otherwise.
+     */
+    public void setSuccess(Boolean success)
+    {
+        this.isSuccess = success;
+    }
+
+    /**
+     * @return get the status code associated with this file upload.
+     */
+    public Integer getStatusCode()
+    {
+        return this.statusCode;
+    }
+
+    /**
+     * Set the status code associated with this file upload request
+     * @param statusCode The status code associated with this file upload request
+     */
+    public void setStatusCode(Integer statusCode)
+    {
+        this.statusCode = statusCode;
+    }
+
+    /**
+     * @return get the status description associated with this file upload.
+     */
+    public String getStatusDescription()
+    {
+        return this.statusDescription;
+    }
+
+    /**
+     * Set the status description associated with this file upload request
+     * @param statusDescription The status description associated with this file upload request
+     */
+    public void setStatusDescription(String statusDescription)
+    {
+        this.statusDescription = statusDescription;
+    }
+
+    /**
+     * Empty constructor: Used only to keep GSON happy.
+     */
```

Comment:
do you need a @SuppressWarnings("unused") for this constructor?

Reply:
This repo doesn't really pay much attention to warnings, so no